lua/starfall/libs_sh/color.lua

```diff
@@ -33,17 +60,31 @@ end
 --- Creates a table struct that resembles a Color
 -- @name builtins_library.Color
 -- @class function
--- @param r - Red
+-- @param r - Red or string hexadecimal color
 -- @param g - Green
 -- @param b - Blue
 -- @param a - Alpha
 -- @return New color
-instance.env.Color = function (r, g, b, a)
-	if r~=nil then checkluatype(r, TYPE_NUMBER) else r = 255 end
-	if g~=nil then checkluatype(g, TYPE_NUMBER) else g = 255 end
-	if b~=nil then checkluatype(b, TYPE_NUMBER) else b = 255 end
-	if a~=nil then checkluatype(a, TYPE_NUMBER) else a = 255 end
-	return wrap({ r, g, b, a })
+function instance.env.Color(r, g, b, a)
+	if isstring(r) then
+		if r[1] == "#" then r = string.sub(r, 2) end
+		if string.match(r, "%X") then
+			SF.Throw("Invalid characters in hexadecimal color", 2)
+		else
+			local h2r = hex_to_rgb[#r]
+			if h2r then
+				return wrap(h2r(r))
+			else
+				SF.Throw("Invalid hexadecimal color length", 2)
+			end
+		end
```

Comment:
```lua
local rgba = tonumber(string.match("#?(%x+)") or SF.Throw("Invalid hex number!",2), 16)
local r,g,b,a
if rgba < 0x1000000 then
    r,g,b,a = math.floor(rgba / 0x1000000)%0x100, math.floor(rgba / 0x10000)%0x100, math.floor(rgba / 0x100)%0x100, 0xFF
else
    r,g,b,a = math.floor(rgba / 0x100000000)%0x100, math.floor(rgba / 0x1000000)%0x100, math.floor(rgba / 0x10000)%0x100, math.floor(rgba / 0x100)%0x100
end
```

Reply:
Actually this won't work for rgba case if red is 0

Reply:
@thegrb93, how about this https://github.com/thegrb93/StarfallEx/pull/1088/commits/2e886c0e464c3aa82021618029fbdf62f15aaa67?